### ISSUE TITLE
Metadata Upload / Regions Not Extracted

### ISSUE THREAD

**Issue by pjdufour** [GeoNode/geonode#2237]
It seems that regions are not being extracted properly from a metadata xml when a .xml file is included in a shapefile upload, such as from:

```xml
       <gmd:descriptiveKeywords>
         <gmd:MD_Keywords>
           <gmd:keyword>
             <gco:CharacterString>Iraq</gco:CharacterString>
           </gmd:keyword>
           <gmd:type>
              <gmd:MD_KeywordTypeCode codeSpace="ISOTC211/19115" codeList="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#MD_KeywordTypeCode" codeListValue="place">place</gmd:MD_KeywordTypeCode>
            </gmd:type>
         </gmd:MD_Keywords>
       </gmd:descriptiveKeywords>
```

**Comment by tomkralidis** [comment]
@pjdufour looks like this happens on [layer upload](https://github.com/GeoNode/geonode/blob/master/geonode/layers/utils.py#L412).  In the [ISO case](https://github.com/GeoNode/geonode/blob/master/geonode/layers/metadata.py#L98), we pick up the **first** set of keywords and set accordingly.

- do we want to house all keyword sets?  If yes, for DC/FGDC output we can probably flatten them
- how can we property extract region keywords?
  - look for keywords w/ keyword type=`place` and set as 1..n `Region` objects?

**Comment by pjdufour** [comment]
So we'd need to throw a "for loop" into https://github.com/GeoNode/geonode/blob/master/geonode/layers/metadata.py#L98 to get at all the keywords (place and theme)?  It doesn't look like GeoNode is outputing keywords for FGDC metadata right now.  Is that a bug or a feature we haven't implemented yet?

Yeah, I think that would be the idea for how to extract.  Check if place keyword matches existing regions.  If can't find a match, then just add as normal keyword.

**Pseudo code**
```Python
regions = extract_keywords(codeListValue="place')
keywords = extract_keywords(codeListValue="theme')
regions_resolved, regions_remainder = resolve_regions(regions_unresolved)
regions_resolved = deduplicate(regions_resolved)
keywords.extend(regions_remainder)
layer.regions.add(regions_resolved)
layer.keywords.add(keywords)
```

**Comment by pjdufour** [comment]
@tomkralidis, added support for ISO metadata XML in #2238.  Would the code be easy to do the same for FGDC and the other formats OR should we bump a wider implementation to 2.5.x?

**Comment by tomkralidis** [comment]
@pjdufour merged via #2241 2241
